xla/service/gpu/gemm_rewriter.cc

```diff
@@ -2314,7 +2325,7 @@ class GemmWorkspaceRewriteVisitor : public DfsHloRewriteVisitor {
       custom_call->set_output_to_operand_aliasing({{{0}, {2, {}}}});
     }
 
-    if (has_aux_output) {
+    if (instr->shape().IsTuple()) {
       for (auto user : instr->users()) {
```

Comment:
Can you replace this entire for-loop with `ReplaceInstr(instr, new_call)`? That should cause all the GetTupleElement instructions to refer to new_call insteade of instr.

I just noticed this but it's unrelated to the fix. So if this is replaced by `ReplaceInstr(instr, new_call)`, it should be done in a separate PR

Reply:
Yeah, let's do it in a separate PR.